pkg/deploy/generator/generator.go

```diff
@@ -0,0 +1,39 @@
+package generator
+
+import (
+	"fmt"
+	"reflect"
+
+	kapi "k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/kubectl"
+	"k8s.io/kubernetes/pkg/runtime"
+
+	deployapi "github.com/openshift/origin/pkg/deploy/api"
+)
+
+var basic = kubectl.BasicReplicationController{}
+
+type BasicDeploymentConfigController struct{}
+
+func (BasicDeploymentConfigController) ParamNames() []kubectl.GeneratorParam {
+	return basic.ParamNames()
+}
+
+func (BasicDeploymentConfigController) Generate(genericParams map[string]string) (runtime.Object, error) {
+	obj, err := basic.Generate(genericParams)
+	if err != nil {
+		return nil, err
+	}
+	switch t := obj.(type) {
+	case *kapi.ReplicationController:
+		obj = &deployapi.DeploymentConfig{
+			ObjectMeta: t.ObjectMeta,
```

Comment:
No, our default is always config for users.  If you don't want triggers
(rare) pass an empty map.

On Aug 21, 2015, at 8:40 AM, David Eads <notifications@github.com> wrote:

In pkg/deploy/generator/generator.go
<https://github.com/openshift/origin/pull/4299#discussion_r37629970>:

> +
> +type BasicDeploymentConfigController struct{}
> +
> +func (BasicDeploymentConfigController) ParamNames() []kubectl.GeneratorParam {
> +	return basic.ParamNames()
> +}
> +
> +func (BasicDeploymentConfigController) Generate(genericParams map[string]string) (runtime.Object, error) {
> +	obj, err := basic.Generate(genericParams)
> +	if err != nil {
> +		return nil, err
> +	}
> +	switch t := obj.(type) {
> +	case *kapi.ReplicationController:
> +		obj = &deployapi.DeploymentConfig{
> +			ObjectMeta: t.ObjectMeta,

If we want this to have a particular trigger, I think I'd want to add it
here rather than setting a defaulter.

—
Reply to this email directly or view it on GitHub
<https://github.com/openshift/origin/pull/4299/files#r37629970>.